src/avif.c

```diff
@@ -721,6 +721,51 @@ avifBool avifCleanApertureBoxConvertCropRect(avifCleanApertureBox * clap,
     return AVIF_TRUE;
 }
 
+// ---------------------------------------------------------------------------
+
+avifBool avifAreGridDimensionsValid(avifPixelFormat yuvFormat, uint32_t imageW, uint32_t imageH, uint32_t tileW, uint32_t tileH, avifDiagnostics * diag)
+{
+    if ((imageW == 0) || (imageH == 0) || (tileW == 0) || (tileH == 0)) {
+        avifDiagnosticsPrintf(diag, "[Strict] Tile and image width and height must be nonzero");
+        return AVIF_FALSE;
+    }
+
+    // ISO/IEC 23000-22:2019, Section 7.3.11.4.2:
+    //   - the tile_width shall be greater than or equal to 64, and should be a multiple of 64
+    //   - the tile_height shall be greater than or equal to 64, and should be a multiple of 64
+    // The "should" part is ignored here.
+    if ((tileW < 64) || (tileH < 64)) {
+        avifDiagnosticsPrintf(diag,
+                              "[Strict] Grid image tiles are smaller than 64x64 (%ux%u). See MIAF (ISO/IEC 23000-22:2019), Section 7.3.11.4.2",
+                              tileW,
+                              tileH);
+        return AVIF_FALSE;
+    }
+
+    // ISO/IEC 23000-22:2019, Section 7.3.11.4.2:
+    //   - when the images are in the 4:2:2 chroma sampling format the horizontal tile offsets and widths, and the output width, shall be even numbers;
+    //   - when the images are in the 4:2:0 chroma sampling format both the horizontal and vertical tile offsets and widths, and the output width and height, shall be even numbers.
```

Comment:
I seem to remember that I analyzed these requirements and concluded that the requirements on imageW and imageH (i.e., output width and height) were overly strict. Is this the problem you reported in the team meeting? If so, we can choose to not enforce the requirements on imageW and imageH and document our decision in the comment.

Note: In fact, it seems that even the requirements on tileW and tileH can be relaxed when the image grid has only one row or one column. For example, if the image grid has only one row, then it seems that tileH can be an odd number when chroma is subsampled vertically.

Reply:
The requirements on tileW and tileH are now relaxed when the image grid has only one row or one column. See the comments I added for details on why it cannot easily be relaxed further.